src/main/java/com/faforever/client/map/MapVaultController.java

```diff
@@ -313,9 +321,12 @@ public void showMoreOwnedMaps() {
     displayMapsFromSupplier(() -> mapService.getOwnedMaps(currentPlayer.getId(), LOAD_MORE_COUNT, ++currentPage));
   }
 
-  private void appendSearchResult(List<MapBean> maps, Pane pane) {
+  private void replaceSearchResult(List<MapBean> maps, Pane pane) {
     JavaFxUtil.assertBackgroundThread();
 
+    Platform.runLater(() -> {
+      pane.getChildren().clear();
+    });
     ObservableList<Node> children = pane.getChildren();
```

Comment:
Please just fill this variable in the runLater part because it is needed there

Reply:
You want me to remove the duplication of '`pane.getChildren?` Fillinf the list happens in batches so i cannot clear there.

Reply:
I want 
```
pane.getChildren().clear();
ObservableList<Node> children = pane.getChildren();
```
in line 341

Reply:
Well its not needed i think (reference assignments are thread-safe), but i think the line `ObservableList<Node> children = pane.getChildren();` should be removed entirely. As far as i can see it does not help to make any code clearer or better, and its at an awkward place should this method get refactored / split up in the future.